PocketTutor-Team/pockettutor-app
# Feature/lessons overview

## Implement homeScreen with role-based navigation and lesson display

### PR description 

The goal of this PR is to introduces the homeScreen that displays the user's lessons based on the user's profile, provides role-based navigation for Tutors and Students to find a Lesson and provide an Icon in the top bar to navigate to the screen where the user can view its profil information. 

### Summary:

- homeScreen Composable: Displays a list of the user's lessons, a message when no lessons are available, or an error screen when no profile is assigned to the current user and has an Icon on the top bar that allow the user to navigate to the information of its profile.
- Role-based Navigation: The bottom navigation bar dynamically updates based on whether the user is a Tutor or Student.
- Lesson Display: Lessons are presented in a card format, with details like the title, time slot, user's first name, and lesson price but design remain to be done by @AlexisAllmnd. Currently the two lessons displayed are hard coded such that one can view the design of a lesson. Once @Philipp0206 is done with #45 it can be link by uncommenting the line "val lessons" in Overview.kt
- Navigation Actions Update: Added support for separate top-level destinations for Tutors and Students in the navigation logic.
- MainActivity Update: Handle different navigation and routing based on the user's role (Student or Tutor) using the NavigationActions object.

Once the ProfilInfoScreen is available we can test the navigation to the information of the profil when clicking on the profil Icon.

### How to test

To test this PR, one can run the MainActivity and once the profil is created, one can check that the screen (Screen.HOME) corresponds to the screenshots given below and when clicking on the "My courses" and "Find a Tutor" the user will nagivate to two different routes that only contains at the moment the home screen (Screen.HOME). The app will crash when we click on the Profil Icon on the Top right as it is already set to navigate to Screen.PROFILE but it requires the composable #43 from @alexs-rufer that is ready in a few hours. 

### Screenshots
Screen.HOME:
<img src="https://github.com/user-attachments/assets/94b84705-30a5-4ec2-aa84-10c638154365" width="300" alt="WhatsApp Image 2024-10-16 at 11:36:07">
